modules/nextflow/src/main/groovy/nextflow/script/WorkflowDef.groovy

```diff
@@ -234,23 +253,174 @@ class WorkflowParamsResolver {
         else
             throw new MissingMethodException(name, WorkflowDef, args)
     }
+}
 
-    private Map argsToMap(Object args) {
-        if( args && args.getClass().isArray() ) {
-            if( ((Object[])args)[0] instanceof Map ) {
-                def map = (Map)((Object[])args)[0]
-                return new HashMap(map)
-            }
+/**
+ * Implements the DSL for executing the workflow
+ *
+ * @author Ben Sherman <bentshermann@gmail.com>
+ */
+@Slf4j
+@CompileStatic
+class WorkflowDsl {
+
+    private Binding binding
+
+    WorkflowDsl(Binding binding) {
+        this.binding = binding
+    }
+
+    @Override
+    Object getProperty(String name) {
+        try {
+            return binding.getProperty(name)
+        }
+        catch( MissingPropertyException e ){
+            return super.getProperty(name)
+        }
+    }
+
+    @Override
+    Object invokeMethod(String name, Object args) {
+        if( name == '_into_topic' ) {
+            final args0 = args as Object[]
+            if( args0[0] instanceof DataflowWriteChannel )
+                _into_topic(args0[0] as DataflowWriteChannel, args0[1] as String)
+            else if( args0[0] instanceof ChannelOut )
+                _into_topic(args0[0] as ChannelOut, args0[1] as String)
+            else
+                throw new IllegalArgumentException("Workflow topic source should be a channel")
```

Comment:
This is ugly but it works. Maybe there is a cleaner way to do it. For me it's a moot point because this stuff will likely be refactored as we separate the language from the runtime